bson/mgocompat/doc.go

```diff
@@ -0,0 +1,54 @@
+// Package mgocompat provides MgoRegistry, a BSON registry compatible with globalsign/mgo's BSON,
+// with some remaining differences. It also provides MgoRegistryRespectNilValues for compatibility
+// with mgo's BSON with RespectNilValues set to true. See the bsoncodec docs for more details on
+// registries.
+//
+// MgoRegistry supports Getter and Setter equivalents by registering hooks. Note that if a value
+// matches the hook for bsoncodec.Marshaler, bsoncodec.ValueMarshaler, or bsoncodec.Proxy, that
+// hook will take priority over the Getter hook. The same is true for the hooks for
+// bsoncodec.Unmarshaler and bsoncodec.ValueUnmarshaler and the Setter hook.
+//
+// The differences between MgoRegistry and globalsign/mgo's BSON library are:
+//
+// 1) MgoRegistry errors instead of silently skipping mismatched types when decoding.
+//
+// 2) MgoRegistry does not have special handling for marshaling array ops ("$in", "$nin", "$all").
+//
+// The driver uses different types than mgo's bson. The differences are:
+//
+// 1) The driver's bson.RawValue is equivalent to mgo's bson.Raw, but uses Value instead
+//    of Data and uses Type, which is a bsontype.Type object that wraps a byte, instead of
+//    bson.Raw's Kind, a byte.
+//
+// 3) The driver uses primitive.ObjectID, which is a [12]byte instead of mgo's
+//    bson.ObjectId, a string. Due to this, the zero value marshals and unmarshals differently
+//    for Extended JSON, with the driver marshaling as `{"ID":"000000000000000000000000"}` and
+//    mgo as `{"Id":""}`. The driver will not unmarshal {"ID":""} or `{"Id":null}` to a
```

Comment:
`{ID: null}` should work after GODRIVER-1175.

Reply:
done